pyteal/ast/maybe.py

```diff
@@ -1,16 +1,17 @@
 from typing import List, Union, TYPE_CHECKING
 
+from pyteal.ast.multi import MultiValue
+
 from ..types import TealType
-from ..ir import TealOp, Op, TealBlock
+from ..ir import Op
 from .expr import Expr
-from .leafexpr import LeafExpr
-from .scratch import ScratchSlot, ScratchLoad
+from .scratch import ScratchLoad
 
 if TYPE_CHECKING:
     from ..compiler import CompileOptions
 
 
-class MaybeValue(LeafExpr):
+class MaybeValue(MultiValue):
```

Comment:
@algoidurovic and reviewers - Trying to understand the thinking:  Do we envision removing `MaybeValue`?

As I understand it, PyTeal retains `MaybeValue` for backwards compatibility.  If that's the case, I'm trying to gauge deprecation policy options.

Reply:
I'd be in favor of deprecating `MaybeValue.hasValue` and `.value` to encourage new users to use the reducer function.

While we have no strict requirement to maintain backwards compatibility between most releases (we are in major version 0), I think we should make a best effort attempt to do so.

Reply:
@jasonpaulos Thanks for the reply.  A few follow up questions:
* If I follow the reply, you're imagining retaining `MaybeValue` without implementation-specific methods.  If I've followed, in what use cases do you envision favoring `MaybeValue` over `MultiValue`?
  * I'm wondering if there's specific use cases you see.  Or if you generally imagine an implementation specific to a single value is helpful.  
  * I'll likely defer to your judgement though I'm wondering about the maintenance cost outweighing the benefits.
 * For any deprecation we undertake, I'm wondering _how_ we enforce the deprecation?
   * A quick google shows a library like https://pypi.org/project/deprecation/.  While that library may not be the way forward, I'm gauging _how_ and _when_ we'd enforce removal of deprecated functionality.

Reply:
My understanding is that we'll keep MaybeValue for the currently supported opcodes (the ticket associated with this PR gives some examples). We can even use MaybeValue for future opcodes if the returned values fit the pattern. Because it now extends the MultiValue class, its features should match MultiValue's so it's not behind feature-wise. This PR changes MaybeValue into a lightweight class that basically just adds a thin layer of specificity on top of MultiValue.

Reply:
@algoidurovic Got it - thanks for the context.  Let's keep as is then - resolving.